File: Visualize_Graph/Script2.R
# Load necessary library
library(ggplot2)

# Read the CSV file
data <- read.csv("Student_Performance.csv")

# Convert the 'Date' column to Date format
data$Date <- as.Date(data$Date, format = "%m/%d/%Y")

data$Month <- format(data$Date, "%Y-%m")
data <- aggregate(Performance_Index ~ Month, data, mean)

# Plotting the line chart
ggplot(data, aes(x = Month, y = Performance_Index, group = 1)) +
  geom_line(color = "red") +
  labs(x = "Date", y = "Performance Index") +
  ggtitle("Performance Index Over Time") +
  theme_minimal()


ggplot(data, aes(x = Month, y = Performance_Index, fill = as.factor(row.names(data)))) +
  geom_bar(stat = "identity") +
  labs(x = "Date", y = "Performance Index") +
  ggtitle("Performance Index Over Time") +
  theme_minimal() +
  scale_fill_manual(values = rainbow(nrow(data))) + 
  #rename legend's title
  guides(fill = guide_legend(title = "Month"))

File: Visualize_Graph/Script2.py
import pandas as pd
import matplotlib.pyplot as plt

# Load the CSV file
data = pd.read_csv('Student_Performance.csv')

# Convert 'Date' column to datetime format
data['Date'] = pd.to_datetime(data['Date'], format='%m/%d/%Y')

# Compute the month for each date
data['Month'] = data['Date'].dt.month

# Compute the average Performance_Index by month
average_performance_by_month = data.groupby('Month')['Performance_Index'].mean().reset_index()

# Plotting the line chart
plt.plot(average_performance_by_month['Month'], average_performance_by_month['Performance_Index'], marker='o', color='green')
plt.xlabel('Month')
plt.ylabel('Average Performance Index')
plt.title('Average Performance Index by Month')
plt.xticks(range(1, 13))  # Assuming data spans all 12 months
plt.grid(True)
plt.show()
